src/types/channels.rs

```diff
@@ -37,24 +41,44 @@ impl Channels {
     ///
     /// ```
     /// # use hls_m3u8::types::Channels;
-    /// let channels = Channels::new(6);
+    /// let channels = Channels::new(6, false);
     ///
     /// println!("CHANNELS=\"{}\"", channels);
     /// # assert_eq!(format!("CHANNELS=\"{}\"", channels), "CHANNELS=\"6\"".to_string());
     /// ```
-    //#[inline]
     #[must_use]
-    pub const fn new(number: u64) -> Self {
-        Self { number }
+    pub const fn new(number: u64, has_joc_content: bool) -> Self {
+        Self {
+            number,
+            has_joc_content,
+        }
     }
 }
 
 impl FromStr for Channels {
     type Err = Error;
 
+    /// Makes a new [`Channels`] struct from a str.
+    ///
+    /// Has significant extra logic to account for the addition of Enhanced AC-3 audio with JOC, which
+    /// allows for strings like "16/JOC".
+    /// #[inline]
     fn from_str(input: &str) -> Result<Self, Self::Err> {
+        // Lots of extra logic to deal with Dolby Atmos
+        let split: Vec<&str> = input.split("/").collect::<Vec<&str>>();
```

Comment:
I think you should use `splitn` here to properly handle erroneous inputs such as "16/JOC/2".